File: cpp/problem_0007/reverse_integer.test.cpp
#define CATCH_CONFIG_MAIN
#include "../catch_amalgamated.hpp"
#include "reverse_integer.cpp"

TEST_CASE("reverse", "[reverse_integer]") {
    Solution solution;
    
    SECTION("Example 1") {
        const int input = 123;
        const int output = 321;
        REQUIRE(solution.reverse(input) == output);
    }
    
    SECTION("Example 2") {
        const int input = -123;
        const int output = -321;
        REQUIRE(solution.reverse(input) == output);
    }
    
    SECTION("Example 3") {
        const int input = 120;
        const int output = 21;
        REQUIRE(solution.reverse(input) == output);
    }
    
    SECTION("Zero") {
        const int input = 0;
        const int output = 0;
        REQUIRE(solution.reverse(input) == output);
    }
    
    SECTION("Single digit positive") {
        const int input = 7;
        const int output = 7;
        REQUIRE(solution.reverse(input) == output);
    }
    
    SECTION("Single digit negative") {
        const int input = -7;
        const int output = -7;
        REQUIRE(solution.reverse(input) == output);
    }
    
    SECTION("Large positive number") {
        const int input = 1534236469;
        const int output = 0; // Should overflow and return 0
        REQUIRE(solution.reverse(input) == output);
    }
    
    SECTION("Large negative number") {
        const int input = -2147447648; // Would reverse to exceed int range
        const int output = 0;
        REQUIRE(solution.reverse(input) == output);
    }
}

File: typescript/problem_0007/ReverseInteger.test.ts
import { test, expect } from 'vitest';
import { reverse } from './ReverseInteger';

test("Reverse Integer 1", () => {
  const input = 123;
  const output = 321;

  expect(reverse(input)).toBe(output);
});

test("Reverse Integer 2", () => {
  const input = -123;
  const output = -321;

  expect(reverse(input)).toBe(output);
});

test("Reverse Integer 3", () => {
  const input = 120;
  const output = 21;

  expect(reverse(input)).toBe(output);
});

test("Reverse Integer 4", () => {
  const input = 0;
  const output = 0;

  expect(reverse(input)).toBe(output);
});
